utils/check_roxie/check_roxie

```diff
@@ -1,11 +1,74 @@
-#!/bin/bash
+#!/usr/bin/python
 
-OUTPUT=`testsocket $1 "<control:alive/>" |grep Status |grep ok -c`
+from __future__ import print_function
+import os
+import re
+import sys
+
+# Output usage if argument is missing
+try:
+        IP = sys.argv[1]
+except IndexError:
+                print("\nNeeds IP address of the target server")
+                print("""
+Usage:
+        check_roxie <ip-address>
+""")
+                sys.exit(1)
+
+status = os.popen("2>/dev/null testsocket %s \"<control:alive/>\"" % IP)
+status_result = status.read()
+roxie_status = re.findall('\<Status\>(.+?)\</Status\>+?', status_result)
+
+# Check if roxie process is up and running
+if roxie_status:
+        if roxie_status[0] == 'ok':
+                # Get state hashes from roxie cluster
+                hashes = []
+                try:
+                        cluster_state = os.popen("2>/dev/null testsocket %s -lock \"<control:state/>\"" % IP)
+                        cluster_state_result = cluster_state.readlines()
+                        for i in cluster_state_result:
+                                if "hash" in i:
+                                        state_hash = re.findall("hash=\"(.*?)\"", i)
+                                        hashes.append(state_hash[0])
+                except:
+                        pass
+
+                total_hashes = len(hashes)
+                same_hash = []
+                different_hash = []
+                # Compare hash state between self and cluster
+                while total_hashes > 0:
+                        local_state = os.popen("2>/dev/null testsocket %s \"<control:state/>\"" % IP)
+                        local_state_result = local_state.read()
+                        my_hash = re.findall("hash='(.*?)'", local_state_result)
+                        for i in hashes:
+                                if my_hash[0] == i:
+                                        same_hash.append(i)
+                                        total_hashes -= 1
+                                elif my_hash[0] != i:
+                                        different_hash.append(i)
```

Comment:
Is it really necessary to collect different hashes rather than count them? I can't see any place in this code where it is used.